lumigator/python/mzai/backend/backend/api/routes/health.py

```diff
@@ -20,19 +21,19 @@ def get_health() -> HealthResponse:
 
 @router.get("/jobs/{job_id}")
 def get_job_metadata(job_id: UUID) -> JobSubmissionResponse:
-    resp = requests.get(f"{settings.RAY_DASHBOARD_URL}/api/jobs/{job_id}")
+    resp = requests.get(urllib.parse.urljoin(settings.ray_jobs, f"{job_id}"))
 
     if resp.status_code == HTTPStatus.NOT_FOUND:
         raise HTTPException(
             status_code=HTTPStatus.NOT_FOUND,
-            detail=f"Job metadata for ID: {job_id} not found",
+            detail=f"Job metadata for ID: {job_id} not found - {resp.text or ''}",
         )
     elif resp.status_code != HTTPStatus.OK:
         loguru.logger.error(f"Unexpected status code getting job metadata text: {resp.status_code}")
         loguru.logger.error(f"Upstream job metadata response: {resp}")
         raise HTTPException(
             status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
-            detail=f"Unexpected error getting job metadata for ID: {job_id}",
+            detail=f"Unexpected error getting job metadata for ID: {job_id} - {resp.text or ''}",
```

Comment:
🤔 same here, we intentionally log the response but we don't just leak it back to the caller.

```suggestion
            detail=f"Unexpected error getting job metadata for ID: {job_id}",
```

Reply:
I can agree for the 404, but in case of an error we are not addressing explicitly isn't it better if we provide extra info in the HTTP response too (and not just the logs)? I just want to be mindful this is not a service but a tool that users will run on their own, and providing more information here might save from having to go through the logs. WDYT?

Reply:
This may depend on the specific setup (private use, service, etc). What about settings to log and/or send back the problem details?

Reply:
404 stays mapped to 404, 400 will translate to 500.